cve_bin_tool/checkers/wget.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+"""
+CVE checker for Wget
+References:
+https://www.cvedetails.com/vulnerability-list/vendor_id-72/product_id-332/GNU-Wget.html
+"""
+import os
+from ..util import regex_find
+
+
+def guess_contains_wget(lines):
+    """Tries to determine if a file includes wget
+    """
+    for line in lines:
+        if "GNU Wget %s, a non-interactive network retriever." in line:
+            return 1
+        if "Please send bug reports and questions to <bug-wget@gnu.org>." in line:
+            return 1
+        if "DEBUG output created by Wget %s on %s." in line:
+            return 1
+    return 0
+
+
+def get_version(lines, filename):
+    """
+    returns version information for wget found in given file.
+    VPkg: gnu, wget
+    """
+    regex = [r"wget: ([0-9]+\.[0-9]+\.[0-9]+)"]
```

Comment:
Be warned: this regex does not detect wget on Ubuntu because there's no such version string.  The only version string I see is the gcc line, which may not work on other distros' versions of wget.

Reply:
Can you please help me with getting an idea about how the version string actually looks like on Ubuntu?

Reply:
@SaurabhK122 

I'm seeing things like:

```
GNU Wget %s built on %s.
GNU Wget %s, a non-interactive network retriever.
```

Also,

```console
$ strings $(which wget) | grep -E '[0-9]+\.[0-9]+\.[0-9]+'
libssl.so.1.0.0
libcrypto.so.1.0.0
GLIBC_2.3.4
GLIBC_2.2.5
1.2.11
1.20.3
GCC: (Solus) 8.2.0
GCC: (Solus) 8.3.1 20190320
```

We might need to add some supplemental data, for instance parsing https://ftp.gnu.org/gnu/wget/ to see what versions there are and then if we see both a version in the list of versions that exist and one of the `GNU Wget %s built on %s.` strings then we know. Thoughts @terriko? Or maybe we should just table this.

Reply:
Looks like the fedora package doesn't have this issue. The strings above ^ were on solus linux, which isn't used much. Its probably safe to merge this.